_posts/2015-10-29-Rust-1.4.md

```diff
@@ -0,0 +1,225 @@
+warn on any code that violates the new rules, but still compile. These warnings
+will turn into errors in Rust 1.5. However, given the train model, the
+community has had time to deal with these changes while 1.4 was in beta, and
+the small number of crates we were aware of have already been fixed.
+
+These soundness fixes enable the return of the ‘scoped threads’ feature, in
+which you can create threads that reference data stored on the stack in a
+safe manner. A few crates have implemented this feature, most notably
+[crossbeam] and [scoped_threadpool]. See their documentation for more
+information.
+
+[crossbeam]: https://crates.io/crates/crossbeam
+[scoped_threadpool]: https://crates.io/crates/scoped_threadpool
+
+[RFC 1212](https://github.com/rust-lang/rfcs/blob/master/text/1212-line-endings.md)
+is also in this release, which changes all functions dealing with reading
+‘lines’ to treat both `\n` and `\r\n` as a valid line-ending. This was
+determined during the RFC process to be a bugfix, but we’re mentioning it
+here to raise awareness. The older behavior of only dealing with `\n` made
+for surprising behavior, where your crate would work well on Linux and Mac OS
+X, but fail on Windows. This fix brings these functions more in-line (😉)
+with expectations.
+
+Rust 1.4 marks an upgrade in our Windows support: Windows builds targeting the
+64-bit MSVC ABI and linker (instead of GNU) are now supported and recommended
+for general use, and will appear on the downloads page for the first time.
+Thank you to all who have helped us work out the kinks since support initially
+landed in Rust 1.2.
+
+Here’s a summary of library changes:
+
+* 48 APIs were stabilized.
+* Eight APIs were deprecated.
+* Two were made faster.
+* Over ten various types implement new traits.
+
+See the [release notes][libnotes] for exact details.
+
+[libnotes]: https://github.com/brson/rust/blob/relnotes/RELEASES.md#libraries
+
+The compiler [no longer uses
+`morestack`](https://github.com/rust-lang/rust/pull/27338), which was a
+holdover implementation detail from long, long ago. We now use guard pages
+and stack probes instead, though stack probes are only implemented on Windows
+so far.
+
+Finally, one major Cargo improvement: [`cargo update` will now print extra
+information about what it is
+changing.](https://github.com/rust-lang/cargo/pull/1931) For example:
+
+```text
+$ cargo update
+    Updating registry `https://github.com/rust-lang/crates.io-index`
+    Updating libc v0.1.8 -> v0.1.10
+    Updating memchr v0.1.3 -> v0.1.5
+    Updating num v0.1.26 -> v0.1.27
+    Updating rand v0.3.9 -> v0.3.10
+    Updating rustc-serialize v0.3.15 -> v0.3.16
+```
+
+Before, it would do this job silently.
+
+### Contributors to 1.4
+
+Rust is a community-driven project, and we're very appreciative of the work of
+the 128 contributors who made 1.4 happen. Thank you!
+
+- Adam Crume
+- Aidan Hobson Sayers
+- Aleksey Kladov
+- Alex Burka
+- Alex Burka
```

Comment:
They were distinct email addresses, but I've asked and they are in fact the same person, two emails.